src/libcore/iter/traits/iterator.rs

```diff
@@ -1504,30 +1504,23 @@ pub trait Iterator {
     /// assert_eq!(odd, vec![1, 3]);
     /// ```
     #[stable(feature = "rust1", since = "1.0.0")]
-    fn partition<B, F>(self, f: F) -> (B, B) where
+    fn partition<B, F>(self, mut predicate: F) -> (B, B) where
         Self: Sized,
         B: Default + Extend<Self::Item>,
         F: FnMut(&Self::Item) -> bool
     {
-        #[inline]
-        fn extend<'a, T, B: Extend<T>>(
```

Comment:
> You call this an improvement, but do you have anything to quantify that?
> 
> In theory I could see this helping performance if the left container type implements a folding extend, combined with an iterator that benefits from folds like `Chain` or `FlatMap`. It would be nice to demonstrate this with a benchmark though.

No; this is just based on published best-practice for iterators. In particular I'm not sure about the partitioner, because the main benefit of using `lhs.extend` is pre-allocation, which only works when you have a known length.

However, it should be an improvement in most cases on `unzip`, because that doesn't change the length of the iterator and will therefore give containers the opportunity to correctly pre-allocate storage

Reply:
Ah, I didn't realize that. Fixed.